R/teal_card.R

```diff
@@ -309,3 +310,35 @@ code_chunk <- function(code, ..., lang = "R") {
   )
   do.call(teal_card, args = elems)
 }
+
+#' Internal helper for `teal_card`` input conversion
+#'
+#' Converts input values to a format compatible with `teal_card`.
+#' This function is used internally to handle common inputs, such as `ggplot` objects,
+#' ensuring they are appropriately converted to evaluable output blocks that can 
+#' be saved to `RDS` file efficiently.
+#'
+#' This function performs the following conversions:
+#' - `ggplot` objects are converted to `recordedplot` objects.
+#' 
+#' If the R option `teal.reporter.disable_teal_card_conversion` is set to `TRUE`, 
+#' no conversion is applied.
+#'
+#' @param x (`object`) An object to be converted. 
+#'
+#' @return The processed object, possibly converted or left unchanged.
+#'
+#' @keywords internal
+.convert_teal_card_input <- function(x) {
+  if (isTRUE(getOption("teal.reporter.disable_teal_card_conversion"))) {
+    return(x)
+  }
+  if (inherits(x, "chunk_output")) {
+    structure(list(.convert_teal_card_input(x[[1]])), class = c("chunk_output"))
+  } else  if (inherits(x, "ggplot")) {
+    code <- deparse1(quote(print(x)))
+    evaluate::evaluate(code)[[2]]
```

Comment:
I'd prefer using `grDevices::recordPlot()` directly as `evaluate::evaluate` does much than just handling a graphic.

Reply:
I think it is a little overkill to use the `evaluate::evaluate` which: 
- parses the code, 
- checks for console-outputs, 
- checks for warnings 
... only to watch for plots. 

https://github.com/r-lib/evaluate/blob/bbd9ddbbb30ab8da56a570487095c2917caacc3d/R/watchout.R#L186

This below `summarizes` the way how evaluate handles plots
```r
{
  if (requireNamespace("ragg", quietly = TRUE) && exists("agg_record", getNamespace("ragg"))) {
    ragg::agg_record()
  } else {
    pdf(file = NULL)
  }
  dev.control(displaylist = "enable")
  dev <- dev.cur()
  on.exit(dev.off(dev))
  print(<block>)
  recordPlot()
}
```

Reply:
That looks good and it works just fine, the ragg functionality that avoids the render might justify adding it to Imports.

This will make it 22 packages in imports by my count.

Reply:
we can skip the `ragg`. I don't mind the imports note as recursive imports length is more important.

Reply:
Removed `ragg` on 0367c29